### DISCUSSION TITLE
Can I check if the metamask mobile app is installed?

### DISCUSSION THREAD

**Discussion by MetaAhn** [WalletConnect/walletconnect-monorepo#1620] [Developer Support] [answered]
I am using WalletConnect v1.0.
I am developing a program using react.
If the device is mobile (Android, iOS), I want to check if the Metamask mobile application is installed.
Do you support that feature?

If not, can you give me some advice on how to solve this problem?

**Comment by pyncz** [comment, marked as answer]
Hi! I'm not from the WalletConnect team, but will try to answer.

I don't think the information about installed apps is available, especially in web. **But** you probably don't need it.
The key thing is that the user chooses the suitable wallet **by himself**, be it MetaMask or any other wallet that supports WalletConnect. So, you can just use [web3modal](https://github.com/WalletConnect/web3modal) or one of the WalletConnect clients, generate a connect link with it and render it as a QR code, so user will be able to scan it & connect with whatever wallet they want.

Be wallet-agnostic, don't rely just on MetaMask - there are plenty wallets for different needs.
